iosHyperskillApp/iosHyperskillApp/Sources/Modules/Profile/Views/Badges/ProfileBadgesGridItemView.swift

```diff
@@ -0,0 +1,145 @@
+import NukeUI
+import shared
+import SwiftUI
+
+extension ProfileBadgesGridItemView {
+    struct Appearance {
+        let badgeIconSize = CGSize(width: 73, height: 86)
+        let lockIconWidthHeight: CGFloat = 12
+        let levelContainerSpacing: CGFloat = 4
+        let cornerRadius: CGFloat = 8
+        let backgroundColor = Color(ColorPalette.surface)
+    }
+}
+
+struct ProfileBadgesGridItemView: View {
+    private(set) var appearance = Appearance()
+
+    let badge: BadgesViewState.BadgeViewState
+
+    let onBadgeTapped: () -> Void
+
+    var body: some View {
+        Button(action: onBadgeTapped) {
+            VStack(spacing: LayoutInsets.defaultInset) {
+                Text(badge.title)
+                    .font(.subheadline)
+                    .foregroundColor(.primaryText)
+
+                buildImageView()
+                    .frame(size: appearance.badgeIconSize)
+
+                VStack(spacing: LayoutInsets.smallInset) {
+                    HStack(spacing: appearance.levelContainerSpacing) {
+                        Text(badge.formattedCurrentLevel)
+                            .font(.caption2)
+                            .foregroundColor(.primaryText)
+
+                        Spacer()
+
+                        if let nextLevel = badge.nextLevel {
+                            Image(systemName: "lock")
+                                .resizable()
+                                .renderingMode(.template)
+                                .foregroundColor(.disabledText)
+                                .aspectRatio(contentMode: .fit)
+                                .frame(widthHeight: appearance.lockIconWidthHeight)
+
+                            Text("\(nextLevel)")
+                                .font(.caption2)
+                                .foregroundColor(.disabledText)
+                        }
+                    }
+
+                    LinearGradientProgressView(progress: badge.progress)
+                }
+            }
+            .padding()
+            .background(appearance.backgroundColor)
+            .cornerRadius(appearance.cornerRadius)
+        }
+    }
+
+    @ViewBuilder
+    private func buildImageView() -> some View {
+        switch BadgesViewStateBadgeImageKs(badge.image) {
+        case .locked:
+            Image(badge.kind.lockedImage)
+                .renderingMode(.original)
+                .resizable()
+        case .remote(let remoteImage):
+            LazyImage(
+                source: remoteImage.previewSource,
```

Comment:
As far as I remember we have preview and full-size images. So, a source for the `LazyImage` should be chosen by these steps:
1. If full-size image cached -> source full-size
2. If preview image is cached -> source preview -> start loading full-size (on done replace source)
3. If no cache -> skeleton animation -> start loading preview (on done follow 2 step)

Reply:
@ivan-magda 
The full-size image is assumed to be used in the bottom sheet. The preview image should be used for the grid in the profile.

Reply:
@ivan-magda I agree with @XanderZhu, full-size images are to big to show them in the grid view and preview source seems to be ok for that (you can compare preview icons on the first screenshot and full source icons on the second one). If we load only preview icons I should add skeletons and make sure that cached image is used if can be, shouldn't I, @ivan-magda ?
![Simulator Screen Shot - iPhone 14 Pro - 2023-08-01 at 12 46 59](https://github.com/hyperskill/mobile-app/assets/30647936/3d050e74-cb7d-47f6-9cc4-4808ccb25dc3)
![Simulator Screen Shot - iPhone 14 Pro - 2023-08-01 at 12 46 13](https://github.com/hyperskill/mobile-app/assets/30647936/a44d6e8e-0851-48a4-bb04-1cf675d2ce19)

Reply:
@XanderZhu @vladkash preview images look very bad - I see graininess. Also, don't forget about high-density devices such as iPads, on those devices it will look more ugly.